packages/compile-common/src/compilerSupplier/strategy.ts

```diff
@@ -0,0 +1,106 @@
+import type { Constructor as FreeConstructor } from "./constructor";
+import type { Results as FreeResults } from "./results";
+
+export namespace Strategy {
+  /**
+   * Type-level description of a compiler supplier strategy
+   */
+  export type Specification = {
```

Comment:
All of these fields need to have their meaning documented in comments.  IMO, this should include which ones are "real" types, which ones are just flags for manually-programmed behavior, and which ones are a mix, but Nick denies that this is a meaningful distinction. :P  Regardless, they all should have their meaning documented in comments.

Reply:
Included docstrings for all fields in all specifications.